x/ecocredit/core/genesis_test.go

```diff
@@ -180,352 +158,228 @@ func TestGenesisValidate(t *testing.T) {
-		},
-		{
-			"expect error: supply is missing",
-			func() json.RawMessage {
-				return json.RawMessage(`{
-					"regen.ecocredit.v1.Class":[{
-						"id":"C01",
-						"admin":"PPUOsQeEHJyQV0ABQzU91iytr9s=",
-						"credit_type_abbrev":"C"
-					}],
-					"regen.ecocredit.v1.Project":[{
-						"id":"P01",
-						"admin":"PPUOsQeEHJyQV0ABQzU91iytr9s=",
-						"class_key":"1",
-						"project_jurisdiction":"AQ"
-					}],
-					"regen.ecocredit.v1.Batch":[{
-						"issuer":"PPUOsQeEHJyQV0ABQzU91iytr9s=",
-						"project_key":"1",
-						"denom":"C01-00000000-00000000-001",
-						"start_date":"2021-04-08T10:40:10.774108556Z",
-						"end_date":"2022-04-08T10:40:10.774108556Z"
-					}],
-					"regen.ecocredit.v1.CreditType": [
-						{
-							"name":"carbon",
-							"abbreviation": "C",
-							"unit":"metric ton C02 equivalent",
-							"precision":6
-						},
-						{
-							"name":"biodiversity",
-							"abbreviation": "BIO",
-							"unit":"acres",
-							"precision":6
-						}
-					],
-					"regen.ecocredit.v1.BatchBalance":[{
-						"batch_key":"1",
-						"address":"mAAyikSMAfVwmlW4BPV2Q6GmpHc=",
-						"tradable":"400.456"
-					}]
-				}`)
+			"expect error: balances are missing",
+			func(ctx context.Context, ss api.StateStore) {
+				require.NoError(t, ss.CreditTypeTable().Insert(ctx, &api.CreditType{
+					Abbreviation: "C",
+					Name:         "carbon",
+					Unit:         "metric ton C02 equivalent",
+					Precision:    6,
+				}))
+				denom := "C01-00000000-00000000-001"
+				key, err := ss.ClassTable().InsertReturningID(ctx, &api.Class{
+					Id:               "C01",
+					Admin:            dummyAddr,
+					CreditTypeAbbrev: "C",
+				})
+				require.NoError(t, err)
+
+				pKey, err := ss.ProjectTable().InsertReturningID(ctx, &api.Project{
+					Id:                  "P01",
+					Admin:               dummyAddr,
+					ClassKey:            key,
+					ProjectJurisdiction: "AQ",
+				})
+				require.NoError(t, err)
+				bKey, err := ss.BatchTable().InsertReturningID(ctx, &api.Batch{
+					Issuer:       dummyAddr,
+					ProjectKey:   pKey,
+					Denom:        denom,
+					StartDate:    &timestamppb.Timestamp{Seconds: 100},
+					EndDate:      &timestamppb.Timestamp{Seconds: 101},
+					IssuanceDate: &timestamppb.Timestamp{Seconds: 400},
+				})
+				require.NoError(t, err)
+				require.NoError(t, ss.BatchSupplyTable().Insert(ctx, &api.BatchSupply{
+					BatchKey:       bKey,
+					TradableAmount: "400.456",
+				}))
 			},
 			defaultParams,
 			true,
-			"supply is not found",
+			"no balances were found",
 		},
 		{
 			"expect error: invalid supply",
-			func() json.RawMessage {
-				return json.RawMessage(`{
-					"regen.ecocredit.v1.CreditType": [
-						{
-							"name":"carbon",
-							"abbreviation": "C",
-							"unit":"metric ton C02 equivalent",
-							"precision":6
-						},
-						{
-							"name":"biodiversity",
-							"abbreviation": "BIO",
-							"unit":"acres",
-							"precision":6
-						}
-					],
```

Comment:
It looks like the previous tests didn't utilize the second credit type but the new tests exclude it completely. Might be nice to add a test case with multiple credit types.

Reply:
See https://github.com/regen-network/regen-ledger/pull/1055#discussion_r861884384